helix-term/src/commands/typed.rs

```diff
@@ -334,6 +334,10 @@ fn write_impl(
     let (view, doc) = current!(cx.editor);
     let path = path.map(AsRef::as_ref);
 
+    if doc.detect_readonly() && !force {
```

Comment:
this check should only fire if `path.is_none()` 
```suggestion
    if doc.detect_readonly() && !force $$path.is_none() {
```

That means that `:w` will fail but `:w <path>` won't fail (as that path is likely not read-only)